Kernel/Memory/VMObject.cpp

```diff
@@ -28,14 +30,23 @@ ErrorOr<FixedArray<RefPtr<PhysicalPage>>> VMObject::try_create_physical_pages(si
 }
 
 VMObject::VMObject(FixedArray<RefPtr<PhysicalPage>>&& new_physical_pages)
-    : m_physical_pages(move(new_physical_pages))
+    : m_physical_pages_count(new_physical_pages.size())
 {
+    m_physical_pages.with([&](auto& array) {
+        FixedArray<RefPtr<PhysicalPage>>::move_elements_to_new_array_from_old_array(array, move(new_physical_pages));
+    });
     all_instances().with([&](auto& list) { list.append(*this); });
 }
 
+void VMObject::will_be_destroyed()
+{
+    m_regions.with([&](auto& list) {
+        VERIFY(list.is_empty());
+    });
+}
+
 VMObject::~VMObject()
 {
-    VERIFY(m_regions.is_empty());
```

Comment:
This assert should still be possible, what's the logic on removing it?

Reply:
We need to grab the lock on `m_regions` to do things on it, right? I do the assert in `VMObject::will_be_destroyed()`, where it's safe to grab the lock. When we destruct things, I'm not sure if it is actually safe to take the lock.

Reply:
Why would it not be safe? You are int the destructor, so nothing has been destroyed yet, unless you explicitly tore it down. The destructor order is strictly specified by the standard. See "Destructor sequence" https://en.cppreference.com/w/cpp/language/destructor

Reply:
I see. It is fixed now.